analyzers/src/SonarAnalyzer.CSharp/Rules/GetTypeWithIsAssignableFrom.cs

```diff
@@ -136,6 +141,9 @@ private static void CheckForIsAssignableFrom(SonarSyntaxNodeReportingContext con
             }
         }
 
+        private static bool IsUnboundedGenericType(TypeOfExpressionSyntax typeOf) =>
+            typeOf.DescendantNodes().Any(x => x is OmittedTypeArgumentSyntax);
```

Comment:
`DescendantNodes()` is typically an expensive operation. Especially on a method level. Inside a `typeof`, it's fine, as we're sure that it will not contain a large sub-tree. So the implementation is good here.

The question is then, what about nested generic types? Can you have `IDictionary<string, ISet<>>`? Let's add UT

Reply:
I've found a property of `GenericNameSyntax` called `IsUnboundGenericName`, which has a very similar implementation to the one we have drafted together:
```cs
        public bool IsUnboundGenericName
        {
            get
            {
                return this.TypeArgumentList.Arguments.Any(SyntaxKind.OmittedTypeArgument);
            }
        }
```
c.p. https://github.com/dotnet/roslyn/blob/main/src/Compilers/CSharp/Portable/Syntax/GenericNameSyntax.cs#L19